File: 이재진/더맵게.py
import heapq
def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)
    while len(scoville) > 1 and scoville[0] < K:
        a = heapq.heappop(scoville)
        b = heapq.heappop(scoville)
        tmp = a + (b * 2)
        heapq.heappush(scoville, tmp)
        answer += 1
    if scoville[0] < K:
        return -1
    return answer

File: 이재진/더맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        
        int answer = 0;
        PriorityQueue<Integer> q = new PriorityQueue<>();
        for (int s:scoville) {
            q.offer(s);
        }
        while (true) {
            if (q.size() == 1) {
                int x = q.poll();
                if (x >= K) {
                    break;
                } else {
                    return -1;
                }
            }
            
            Boolean flag = false;
            for (int i:q) {
                if (i < K) {
                    flag = true;
                    break;
                }
            }
            if (flag) {
                int tmp = 0;
                tmp += q.poll();
                tmp += (q.poll()*2);
                q.offer(tmp);
                answer++;
            } else {
                break;
            }
        }
        return answer;
    }
}
